contracts/Vault1.sol

```diff
@@ -1,6 +1,26 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault1 {
-    
-}
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault1 is ERC20  {
+
+    mapping(address => uint256) public balance;
+    address public owner;
+
+    constructor() ERC20('DepositCoin', 'DC'){
```

Comment:
This contract should not be an ERC20 token, it should be able to take in any ERC20 token address and work with that